packages/react-data-grid/src/HeaderCell.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import classNames from 'classnames';
+import { isElement } from 'react-is';
+import Draggable from './Draggable';
+import { isFrozen } from './ColumnUtils';
+import { HeaderRowType } from './common/enums';
+import { Column } from './common/types';
+
+function SimpleCellRenderer({ column }: { column: Column }) {
+  const headerText = column.rowType === HeaderRowType.HEADER ? column.name : '';
+  return <div>{headerText}</div>;
+}
+
+interface Props {
+  renderer?: React.ReactElement | React.FunctionComponent<{ column: Column }>;
```

Comment:
How would you write `getCell` then?

Reply:
We can always use `<Renderer {..props}>` but I wonder if we should just use function now

```
renderer?: React.ReactElement |({ column: Column }): ReactNode
```

Wdyt?

Reply:
Well that's what I did already.

Reply:
There is a difference, one is a component and other is a function. 

![image](https://user-images.githubusercontent.com/1452717/55910643-8673a200-5ba4-11e9-86e4-2a134ea3dfa5.png)

Reply:
![image](https://user-images.githubusercontent.com/567105/55910684-edaa4a00-5bd6-11e9-9eb7-9c1a2cdc54a7.png)
The first line indicates that it's a function. It also has extra properties like function components.

Reply:
I understand a component is ultimately a function but we are calling it instead of letting react call it for us. I am not sure if that can cause any issues or not. As @qili26 pointed it may not work with hooks.

Reply:
I'll use JSX notation then.